apps/meteor/tests/end-to-end/api/livechat/04-dashboards.ts

```diff
@@ -777,6 +778,198 @@ describe('LIVECHAT - dashboards', function () {
+			expect(originalFirstResponseTimeInSeconds).to.be.greaterThanOrEqual(firstDelayInS);
+		});
+
+		it('should correctly calculate the average time of first responses for an agent', async () => {
+			const response = await startANewLivechatRoomAndTakeIt({ agent: agent.credentials });
+			roomId = response.room._id;
+
+			await sleep(secondDelayInS * 1000);
+			await sendAgentMessage(roomId, 'first response from agent', agent.credentials);
+
+			const today = moment().startOf('day').format('YYYY-MM-DD');
+			const result = await request
+				.get(api('livechat/analytics/agent-overview'))
+				.query({ from: today, to: today, name: 'Avg_first_response_time' })
+				.set(credentials)
+				.expect('Content-Type', 'application/json')
+				.expect(200);
+
+			expect(result.body).to.have.property('success', true);
+			expect(result.body).to.have.property('head');
+			expect(result.body).to.have.property('data');
+			expect(result.body.data).to.be.an('array');
+
+			const agentData = result.body.data.find(
+				(agentOverviewData: { name: string; value: string }) => agentOverviewData.name === agent.user.username,
+			);
+			expect(agentData).to.not.be.undefined;
+			expect(agentData).to.have.property('name', agent.user.username);
+			expect(agentData).to.have.property('value');
+			const averageFirstResponseTimeInSeconds = moment.duration(agentData.value).asSeconds();
+			expect(averageFirstResponseTimeInSeconds).to.be.greaterThan(originalFirstResponseTimeInSeconds);
+			expect(averageFirstResponseTimeInSeconds).to.be.greaterThanOrEqual((firstDelayInS + secondDelayInS) / 2);
+			expect(averageFirstResponseTimeInSeconds).to.be.lessThan(secondDelayInS);
+		});
+	});
+
+	describe('[livechat/analytics/agent-overview] - Best first response time', () => {
+		let agent: { credentials: Credentials; user: IUser & { username: string } };
+		let originalBestFirstResponseTimeInSeconds: number;
+		let roomId: string;
+
+		before(async () => {
+			agent = await createAnOnlineAgent();
+		});
+
+		after(() => deleteUser(agent.user));
+
+		it('should return no best response time for an agent if no response has been sent in the period', async () => {
+			await startANewLivechatRoomAndTakeIt({ agent: agent.credentials });
+
+			const today = moment().startOf('day').format('YYYY-MM-DD');
+
+			const result = await request
+				.get(api('livechat/analytics/agent-overview'))
+				.query({ from: today, to: today, name: 'Best_first_response_time' })
+				.set(credentials)
+				.expect('Content-Type', 'application/json')
+				.expect(200);
+
+			expect(result.body).to.have.property('success', true);
+			expect(result.body).to.have.property('head');
+			expect(result.body).to.have.property('data');
+			expect(result.body.data).to.be.an('array');
+			expect(result.body.data).to.not.deep.include({ name: agent.user.username });
+		});
+
+		it('should only consider a first response has been sent when an agent sends a text message', async () => {
```

Comment:
This is more of a question: does having the same name for different tests a good idea? Even if they are on different describes, maybe we could make them more unique? Line 815 and 908

Reply:
hmmm good suggestion. I'll change both descriptions and be more specific